### DISCUSSION TITLE
Tasks details are missing for dynamically generated dags

### DISCUSSION THREAD

**Discussion by rajatm91** [apache/airflow#23489] [Q&A] [answered]
Hi Everyone , 
We recently migrated our airflow instance from 1.10.15 to 2.3. One of the new and interesting features is the grid view. I have a question around it. 

Currently, our airflow instance has two types of dags :

1. dags which are dynamically generated from configuration 
2. dags which are generated using static python code.

We observed that the dags which are dynamically generated are missing task details on the UI. 
<img width="666" alt="image" src="https://user-images.githubusercontent.com/12770520/166766929-d6e4514c-1c16-46c1-8f9f-2ac3ffae8f30.png">

while the dag which is generated using the static python code has the task details 
<img width="666" alt="image" src="https://user-images.githubusercontent.com/12770520/166770150-513b79bf-e0e0-4d67-b81a-4c4167b8796e.png">

Not sure if I'm missing something here to show the tasks for the dynamically generated dags. Any help would be appreciated 🙏 .

**Comment by bbovenzi** [comment]
Would you be able to share a sample dag where this happens?

**Reply by rajatm91** [reply]
How should I share the dag?

**Reply by bbovenzi** [reply]
Paste the DAG code here. But strip out any sensitive information.

**Reply by rajatm91** [reply]
to generate the dag, we use this : 
```
dag_factory = TestDag(config_manager)

dags = dag_factory.get_dags()

for dag in dags:
    globals()[dag.dag_id] = dag
```

we have a separate class `AnonymizationDAGFactory`, where the dag and its tasks are defined. I cannot share the exact code but will share a pseudo-code: 
```
class TestDag:
    
    def __init__(self) -> None:
        self.configs_root_dir = <config_path>
        
    def get_dags(self) -> t.List[DAG]:
        config_files = self._get_configuration_files_list()
       
        dags = []

        for config_file_path in config_files:
            pipeline_config = json.load(config_file_path)

            dag = self._create_dag(pipeline_config)
            self._add_tasks_to_dag(
                pipeline_config,
                dag,
                metastore_config
            )

            dags.append(dag)

        return dags

    def _create_dag(self) -> DAG:
        dag_id = "test"

        return DAG(
            dag_id=dag_id,
            schedule_interval="4 * * * *",
            start_date=xxxx,
            max_active_runs=3,
            user_defined_filters={
                "extract_total_seconds": xxxx
            }
        )

    def _get_configuration_files_list(self) -> t.List[str]:
        return [os.path.join(root, f) for root, directory, file in os.walk(self.configs_root_dir)
                for f in file if f.endswith('.dag.json')]

    def _add_tasks_to_dag(
            self,
            pipeline_config: PipelineConfigDto,
            dag: DAG,
            metastore_config: dict) -> None:
        task_factory = TaskFactory(dag, self.config_manager, pipeline_config)

        for task_group_config in pipeline_config.task_groups:
            self._add_task_group_to_dag(task_group_config, metastore_config, task_factory)

    def _add_task_group_to_dag(
            self,
            task_group_config: TaskGroupDto,
            metastore_config: dict,
            task_factory: TaskFactory) -> None:

        task1 = task_factory.task1()

        if task_group_config.task2:
            task2 = task_factory.task2()
        else:
            task2 = None

        if task_group_config.task3:
            task3 = task_factory.task3()

            if task_group_config.task4 :
                task4 = task_factory.task4()
                task5 = task_factory.task5()
                task3 >> task4
                task4 >> task5
        else:
            task3 = None

        if task2:
            task1 >> task2
            if task3:
                task2 >> task3
        elif task3:
            task1 >> task3
```
We have a separate class where all Tasks are defined. But I hope this should give an idea how dags are generated.

**Reply by rajatm91** [reply]
Hi @bbovenzi,
were you able to figure out if it's a bug? Also are there any workarounds for this?

**Reply by bbovenzi** [reply]
I'm still having an issues replicating. I don't have enough info to create your dag factory. So let's try 2 things:

1. Could you check Inspect Element -> console, type in "gridData" and post the response here?
2. Is the graph view working as expected for this DAG?

**Reply by rajatm91** [reply]
>Could you check Inspect Element -> console, type in "gridData" and post the response here?
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/12770520/169111686-eb02b2b5-a42b-4a0c-8b40-6a4213abbb1e.png">
>Is the graph view working as expected for this DAG?

Yes, it is working as expected.

**Reply by bbovenzi** [reply]
Can you click "Auto-refresh" and see what is returned by the `tree_data` network request?

**Reply by rajatm91** [reply]
Response from clicking "Auto-Refresh" :
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/12770520/169119079-91aab838-8d57-4918-8bb9-e49c0143fd8c.png">

I can see the task details in the tree_data response, but the auto-refresh gets disabled automatically even after enabling it.

**Reply by bbovenzi** [reply]
Yes, it will turn off when there are no active dag runs.

And there were no console or webserver errors? This is weird because it seems like the data is there in the `tree_data` response.

**Reply by gmsantos** [reply]
@bbovenzi @rajatm91  the tasks have a `.` in the name, it is probably related to https://github.com/apache/airflow/issues/23580

**Reply by gmsantos** [reply]
It seems to be fixed in Airflow 2.3.1rc1 #23852

**Reply by rajatm91** [reply]
thanks, @gmsantos for pointing it out. Indeed that was the problem.

**Comment by bbovenzi** [comment, marked as answer]
As @gmsantos noted, this was because the generated DAG in question had task names with `.` in the name. That has since been fixed in #23590
